src/App.tsx

```diff
@@ -58,6 +60,31 @@ const BackgroundProvider: FC<BackgroundProviderProps> = ({ children }) => {
   return <>{children}</>;
 };
 
+interface ThemeProviderProps {
+  children: JSX.Element;
+}
+
+const ThemeProvider: FC<ThemeProviderProps> = ({ children }) => {
+  const { theme } = useUrlParams();
+  const [previousTheme, setCurrentTheme] = useState<string | null>(
+    document.body.classList.item(0),
+  );
+  useEffect(() => {
```

Comment:
```suggestion
  useLayoutEffect(() => {
```

So that the UI doesn't flash from dark to light mode when first mounted